src/Authenticator/FormAuthenticator.php

```diff
@@ -81,10 +81,15 @@ protected function _getData(ServerRequestInterface $request)
      */
     protected function _buildLoginUrlErrorResult($request)
     {
+        $uri = $request->getUri();
+        if (property_exists($uri, 'base')) {
+            $uri = $uri->withPath($uri->base . $uri->getPath());
+        }
+
         $errors = [
             sprintf(
                 'Login URL `%s` did not match `%s`.',
-                (string)$request->getUri(),
+                (string)$uri,
```

Comment:
You can change this line to `(string)$request->getAttribute('webroot') . (string)$request->getUri()` and avoid all other code.

Reply:
The thing is that `(string)$request->getUri()` returns an absolute URL. So it would have to use the relative URL instead then, but that would require additional code to extract the host, the scheme, the port, etc.

Reply:
Well then just replace the use of `$uri->base` above with `$request->getAttribute('base')`. That would avoid the need for property existence check.